allennlp/models/encoder_decoders/simple_seq2seq.py

```diff
@@ -135,8 +136,7 @@ def forward(self,  # type: ignore
         else:
             num_decoding_steps = self._max_decoding_steps
         decoder_hidden = final_encoder_output
-        with torch.cuda.device_of(decoder_hidden):
-            decoder_context = Variable(torch.zeros(1, self._decoder_output_dim))
+        decoder_context = Variable(source_mask.data.new().resize_(batch_size, self._decoder_output_dim).fill_(0))
```

Comment:
I see that `resize_` also truncates or expands tensors if needed.

Reply:
Yes, that's the point.